### PR TITLE
Addressing bugs from Joplin QA

### PR DESCRIPTION
**Pull request by briaguya**
* Fix "Save Draft" button on create page https://github.com/cityofaustin/techstack/issues/686
* Fix header styling on create page https://github.com/cityofaustin/techstack/issues/686
* Create Content Modal:
  * Clear data in fields and radio buttons "close" https://github.com/cityofaustin/techstack/issues/696
  * Fix typo https://github.com/cityofaustin/techstack/issues/700
  * Always say "characters remaining" instead of switching from "character limit" https://github.com/cityofaustin/techstack/issues/701
* Add prettier config so we don't fight defaults https://github.com/cityofaustin/techstack/issues/738

### CODE REVIEW

File: joplin/js/CreateContentModal/ButtonBar.js [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
-  })}>
-    <div
-      className="ButtonBar__button"
-      onClick={handleBackButton}
-    >
+const ButtonBar = ({
+  activeStep,
+  handleBackButton,
+  handleNextButton,
+  handleCloseButton
```
Comment (briaguya) [inline joplin/js/CreateContentModal/ButtonBar.js]: This is for https://github.com/cityofaustin/techstack/issues/696

File: joplin/js/CreateContentModal/ButtonBar.js [hunk 2]
```diff
@@ -1,29 +1,32 @@
...
+  >
+    <div className="ButtonBar__button" onClick={handleBackButton}>
       Back
     </div>
-    <button type="button"
+    <button
+      type="button"
       className="ButtonBar__button ButtonBar__button--reset"
       data-dismiss="modal"
+      onClick={handleCloseButton}
```
Comment (briaguya) [inline joplin/js/CreateContentModal/ButtonBar.js]: https://github.com/cityofaustin/techstack/issues/696

File: joplin/js/CreateContentModal/index.js [hunk 3]
```diff
@@ -52,70 +52,87 @@ class CreateContentModal extends Component {
...
     window.location.href = this.state.redirectUrl;
-  }
+  };
 
   writeToLocalStorage = () => {
     localStorage.wagtailCreateModal = JSON.stringify(this.state);
-  }
+  };
 
+  handleCloseButton = e => {
```
Comment (briaguya) [inline joplin/js/CreateContentModal/index.js]: https://github.com/cityofaustin/techstack/issues/696

File: joplin/templates/wagtailadmin/pages/create.html [hunk 4]
```diff
@@ -12,27 +12,14 @@
     <input type="hidden" id="style_guide_url" name="style_guide_url" value="{{ style_guide_url }}">
     <div class="header secondary coa-header--secondary">
         <div>
-            <div class="create_page_header">
-              <div class="left col9 header-title">
-                  <h1 class="icon icon-doc-empty-inverse">
-                  {% trans 'New' %} <span>{{ content_type.model_class.get_verbose_name }}</span></h1>
-              </div>
-            </div>
+            <h1 class="create_page_header">
```
Comment (briaguya) [inline joplin/templates/wagtailadmin/pages/create.html]: https://github.com/cityofaustin/techstack/issues/686

File: joplin/templates/wagtailadmin/pages/create.html [hunk 5]
```diff
@@ -12,27 +12,14 @@
...
-                <ul role="menu">
-                    {% if parent_page_perms.can_publish_subpage %}
-                        <li>
-                            <button type="submit" name="action-publish" value="action-publish" class="button button-longrunning" tabindex="3" data-clicked-text="{% trans 'Publishing…' %}" {% if page.locked %}disabled {% endif %}><span class="icon icon-spinner"></span><em>{% trans 'Publish' %}</em></button>
-                        </li>
-                    {% endif %}
-                    <li><input type="submit" name="action-submit" value="{% trans 'Submit for moderation' %}" class="button" /></li>
-                </ul>
-            </div>
+          <button type="submit" form="page-edit-form" class="button action-save button-longrunning editor__save-button" tabindex="3" data-clicked-text="{% trans 'Saving…' %}"><span class="icon icon-spinner"></span><em>{% trans 'Save draft' %}</em></button>
```
Comment (briaguya) [inline joplin/templates/wagtailadmin/pages/create.html]: https://github.com/cityofaustin/techstack/issues/686

File: joplin/js/CreateContentModal/ChooseTitleStep.js [hunk 6]
```diff
@@ -11,35 +11,36 @@ const ChooseTitleStep = ({
 }) => (
   <div className="CreateContentModal__step">
     <h2 className="CreateContentModal__header">
-      { pageType === 'department' && (
+      {pageType === "department" && (
         <span>
-          Write an the full name of the department, without abrreviations or acroynms.
+          Write the full name of the department without abbreviations or
```
Comment (briaguya) [inline joplin/js/CreateContentModal/ChooseTitleStep.js]: https://github.com/cityofaustin/techstack/issues/700

File: joplin/js/CreateContentModal/ChooseTitleStep.js [hunk 7]
```diff
@@ -11,35 +11,30 @@ const ChooseTitleStep = ({
...
-              className={
-                characterCount > maxCharacterCount
-                  ? "ChooseTitleStep__input-label--right ChooseTitleStep__input-label--red"
-                  : "ChooseTitleStep__input-label--right"
-              }>
-              Characters remaining: { maxCharacterCount - characterCount }
-            </span>
-          )
-      }
+      <span
```
Comment (briaguya) [inline joplin/js/CreateContentModal/ChooseTitleStep.js]: https://github.com/cityofaustin/techstack/issues/701

**Comment by briaguya** [comment]
@toribr per our conversation earlier today I will address the issues with the page header in https://github.com/cityofaustin/techstack/issues/737

File: joplin/package.json [hunk 8]
```diff
@@ -30,5 +30,10 @@
     "react": "^16.4.2",
     "react-dom": "^16.4.2",
     "url-loader": "^1.1.0"
+  },
+  "prettier": {
```
Comment (briaguya) [inline joplin/package.json]: https://github.com/cityofaustin/techstack/issues/738

File: joplin/js/CreateContentModal/ButtonBar.js [hunk 9]
```diff
@@ -1,29 +1,32 @@
...
-    >
+const ButtonBar = ({
+  activeStep,
+  handleBackButton,
+  handleNextButton,
+  handleCloseButton
+}) => (
+  <div
+    className={classNames({
+      ButtonBar: activeStep > 0,
```
Comment (ifsimicoded) [inline joplin/js/CreateContentModal/ButtonBar.js]: This isn't your code Brian, but I don't think we need this conditional so let's simplify the code. The 'ButtonBar' class should always show up. The only thing that needs to be conditional is the ButtonBar--hidden modifier.
```
className={classNames({
  "ButtonBar",
  "ButtonBar--hidden": activeStep === 0
})}
```

File: joplin/js/CreateContentModal/index.js [hunk 10]
```diff
@@ -52,70 +52,87 @@ class CreateContentModal extends Component {
...
                     handleTopicSelect={this.handleTopicSelect}
                     themeTopicTree={THEME_TOPIC_TREE}
                   />
-                }
+                )}
                 <ButtonBar
                   handleBackButton={this.handleBackButton}
                   handleNextButton={this.handleNextButton}
+                  handleCloseButton={this.handleCloseButton}
                   activeStep={this.state.activeStep}
```
Comment (ifsimicoded) [inline joplin/js/CreateContentModal/index.js]: Real big nit, so feel free to ignore. It seems like the only thing that activeStep is used for in ButtonBar is to show or hide the component so we could make this more obvious.
```
<ButtonBar
  handleBackButton={this.handleBackButton}
  handleNextButton={this.handleNextButton}
  handleCloseButton={this.handleCloseButton}
  hideButtonBar={!this.state.activeStep}
/>
```

File: joplin/templates/wagtailadmin/pages/create.html (hunk 4 above)
Comment (ifsimicoded) [inline joplin/templates/wagtailadmin/pages/create.html]: There's a fair amount of sharing between the create and edit pages here. Can these header be turned into it's own template/partial that we include? We can pass in variables to turn off the share, preview and fancy draft buttons.

**Comment by briaguya** [comment]
@ifsimicoded re: the redirect loader process - Since our changes in https://github.com/cityofaustin/techstack/issues/737 mean we're going to be actually saving the newly created page at the end of the modal, It'd probably make sense to add a spinner or something. Any specific ideas on how we want this to look are more than welcome (especially as comments on that techstack issue!)

Thanks for the great feedback!

File: joplin/templates/wagtailadmin/pages/create.html (hunk 4 above)
Comment (ifsimicoded) [reply joplin/templates/wagtailadmin/pages/create.html]: @briaguya will this work? There will be a number of required fields empty? I think making all fields except title and topic optional isn't a great plan either. I'll add this comment in the cityofaustin/techstack#737 too

Comment (ifsimicoded) [reply joplin/templates/wagtailadmin/pages/create.html]: Oh, I see, already noted in  cityofaustin/techstack#737 -- not sure I agree with the approach of only requiring on publish. It's not obvious to the content creator that a field is required until the last minute? I'll this note to the issue.

**Pull request merged**